src/twisted/test/test_defer.py

```diff
@@ -645,6 +645,121 @@ def test_cancelGatherResultsWithAllDeferredsCallback(self):
         self.assertEqual(callbackResult[1], "Callback Result")
 
 
+    def test_unbox(self):
+        """
+        L{defer.unbox} should unbox a Deferred that has already had it's
+        callback function called.
+        """
+        result = object()
+
+        d = defer.Deferred()
+        d.callback(result)
+
+        self.assertIs(defer.unbox(d), result)
+
+
+    def test_unboxRaisesFailureException(self):
+        """
+        L{defer.unbox} should raise the original exception if the result of the
+        deferred is a failure.
+        """
+        d = defer.Deferred()
+        d.errback(ValueError())
+
+        self.assertRaises(ValueError, defer.unbox, d)
+
+        # Without this the test fails because of an unhandled error.
+        d.addErrback(lambda _: None)
+
+    def test_unboxReturnsFailureWithNoRaises(self):
+        """
+        L{defer.unbox} should return the failure instance if raises=False.
+        """
+        d = defer.Deferred()
```

Comment:
Might solve your initial motivation to add `passthrough` if you `def cbDeferred(self): d = Deferred(); d.callback(1234); return d` and `def ebDeferred(self, exception): d = Deferred(); d.addErrback(lambda _: None); d.errback(exception); return d` and did your `lambda _: None` there in the latter so you didn't have to keep repeating it.